packages/playwright-test/src/reporters/json.ts

```diff
@@ -19,13 +19,14 @@ import path from 'path';
 import type { FullConfig, TestCase, Suite, TestResult, TestError, TestStep, FullResult, Location, Reporter, JSONReport, JSONReportSuite, JSONReportSpec, JSONReportTest, JSONReportTestResult, JSONReportTestStep } from '../../types/testReporter';
 import { prepareErrorStack } from './base';
 import { MultiMap } from 'playwright-core/lib/utils/multimap';
+import type { FullConfigInternal } from '../types';
 
 export function toPosixPath(aPath: string): string {
   return aPath.split(path.sep).join(path.posix.sep);
 }
 
 class JSONReporter implements Reporter {
-  config!: FullConfig;
+  config!: FullConfigInternal;
```

Comment:
This suggests that reporters need internal APIs, which does not sound right.

Reply:
If config is used, you have configFile. If config is not used, I don't think you can even configure these reporters.